lib/portlayer/logging/logging.go

```diff
@@ -15,13 +15,123 @@
 package logging
 
 import (
+	"context"
 	"fmt"
+	"sync"
 
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/vmware/govmomi/object"
 	"github.com/vmware/govmomi/vim25/types"
+	"github.com/vmware/vic/lib/portlayer/event/collector/vsphere"
+	"github.com/vmware/vic/lib/portlayer/event/events"
 	"github.com/vmware/vic/lib/portlayer/exec"
+	"github.com/vmware/vic/pkg/backoff"
 	"github.com/vmware/vic/pkg/trace"
 )
 
+var once sync.Once
+
+func Init(ctx context.Context) error {
+	once.Do(func() {
+		// Subscribe to vm events
+		exec.Config.EventManager.Subscribe(
```

Comment:
Currently vic has a very simplistic event handler, so we are unable to subscribe by specific event type -- say *VmGuestShutdownEvent* or *VmPoweredOnEvent* -- so we subscribe for all VM events.  We do this for the exec layer here: https://github.com/vmware/vic/blob/master/lib/portlayer/exec/exec.go#L101 which results in a call back to: https://github.com/vmware/vic/blob/master/lib/portlayer/exec/exec.go#L129 --> that func could be refactored to handle these events as well and just callout to this logging eventCallBack.  This would reduce the number of times that event is passed around.

You can keep this separate subscription if you like but then we'll have 3 different callbacks for VM Events that will fire for all of the events identified in the vsphere/collector.go (https://github.com/vmware/vic/blob/master/lib/portlayer/event/collector/vsphere/collector.go#L164) -- IMO causing undo noise / logging /etc..

I prefer the separation of concerns w/this approach but given the simplistic event handler my approach would be to refactor the existing exec.eventCallback

my $.02